src/lightning/pytorch/CHANGELOG.md

```diff
@@ -168,6 +168,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - The `FSDPStrategy.load_optimizer_state_dict` and `FSDPStrategy.load_model_state_dict` are a no-op now ([#18358](https://github.com/Lightning-AI/lightning/pull/18358))
 
 
+- The `*_step(dataloader_iter)` flavor now no longer takes the `batch_idx` in the signature ([#18390](https://github.com/Lightning-AI/lightning/pull/18390))
```

Comment:
TODO: check if we have an appropriate error message for this

Reply:
There is alternative: We can keep the batch_idx in the signature and it will assume the batch_idx at the time the step method gets called. This would also simplify the internal logic. Of course, we can remove it from all examples since its optional. It would be a reasonable choice IMO.

Reply:
I didn't add any explicit error message or this, but I would suggest that we don't since this API experimental and undocumented.

For the same reason I would not keep `batch_idx` in the signature. I also don't see how it would simplify the internal logic